### PR TITLE
UF-7236: Add SFTP integration

### PR DESCRIPTION
**Pull request by ronnberg**

### CODE REVIEW

File: src/main/java/se/sundsvall/billingpreprocessor/integration/sftp/SftpConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	@Bean
+	@ServiceActivator(inputChannel = "toSftpChannel")
+	public MessageHandler handler(SessionFactory<SftpClient.DirEntry> factory, SftpProperties properties) {
+		SftpMessageHandler handler = new SftpMessageHandler(factory);
+		handler.setRemoteDirectoryExpression(new LiteralExpression(properties.remoteDir()));
+		handler.setFileNameGenerator(message -> {
+			if (message.getPayload() instanceof Resource) {
+				return ((Resource) message.getPayload()).getFilename();
+			} else {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/billingpreprocessor/integration/sftp/SftpConfiguration.java:45]: Redundant else-clause

File: pom.xml [hunk 2]
```diff
@@ -52,6 +56,16 @@
 			<artifactId>jpa</artifactId>
 			<version>${spring-filter.version}</version>
 		</dependency>
+		<dependency>
```
Comment (josc04) [inline pom.xml:59]: Use same version convention as for other dependencies (for example `<version>${spring-filter.version}</version>`)? Comment is valid for both dependencies below.

File: src/main/java/se/sundsvall/billingpreprocessor/integration/sftp/SftpConfiguration.java (hunk 1 above)
Comment (ronnberg) [reply src/main/java/se/sundsvall/billingpreprocessor/integration/sftp/SftpConfiguration.java:45]: Not if we want to foce usage of message payload with this bean. Or are you suggesting to remove the if clause completly and let a ClassCastException be thrown?

Comment (ronnberg) [reply src/main/java/se/sundsvall/billingpreprocessor/integration/sftp/SftpConfiguration.java]: Ok, I thought you meant the entire clause. Will fix

**Comment by foobar-1337** [comment]
Approved! There is also a new codesmell. Don't forget that one... :-)

**Pull request merged**